File: backend-python/src/tools.py
import requests


def get_current_weather(city: str) -> dict:
    # Placeholder function to simulate getting current weather data
    # In a real implementation, this would call a weather API
    print(f"Getting current weather for {city}")
    return {"city": city, "temperature": "25°C", "condition": "Sunny"}


def upload_post(title, description):
    import requests
    from requests_toolbelt.multipart.encoder import MultipartEncoder

    url = "https://node119.cs.colman.ac.il:4000/posts"
    headers = {
        "accept": "application/json, text/plain, */*",
        "accept-language": "en-US,en;q=0.9",
        "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2N2RhZmJmZjEwNDQ4MmNjMDI0MGQ4YzkiLCJyYW5kb20iOiIwLjIxOTA1OTU1NTU1NzA3NjkiLCJ1c2VybmFtZSI6InJveXRoZWtpbmciLCJlbWFpbCI6InJveTMzNDQxY0BnbWFpbC5jb20iLCJpbWFnZSI6Imh0dHBzOi8vbGgzLmdvb2dsZXVzZXJjb250ZW50LmNvbS9hL0FDZzhvY0pRN3MzM1NQX29mdHhFamVETk5kMlNlNVlrdXhiQUxCSFB4aWJ2VkZNZXRoLWpkLV9kZXc9czk2LWMiLCJpYXQiOjE3NDI4NDg1MDksImV4cCI6MTc0Mjg1MjEwOX0.iu-EGPFBJ0QLj0n7_rPsMYbqMQ_mVOusxpvpsqVKAwI",
        "sec-ch-ua": '"Chromium";v="134", "Not:A-Brand";v="24", "Brave";v="134"',
        "sec-ch-ua-mobile": "?0",
        "sec-ch-ua-platform": '"macOS"',
        "sec-fetch-dest": "empty",
        "sec-fetch-mode": "cors",
        "sec-fetch-site": "same-site",
        "sec-gpc": "1",
    }

    s = f'{{"content":"{description}","title":"{title}"}}'

    # Prepare the multipart form data
    multipart_data = MultipartEncoder(
        fields={
            "post": (None, str(s), "application/json"),
            "file": (
                "image.jpeg",
                open("../image.jpeg", "rb"),
                "image/jpeg",
            ),
        }
    )

    headers["Content-Type"] = multipart_data.content_type

    response = requests.post(url, headers=headers, data=multipart_data, verify=False)

File: backend/src/tools.ts
import axios from "axios";
import FormData from "form-data";
import fs from "fs";
import path from "path";

// Simulated weather tool.
export function getCurrentWeather(city: string): {
  city: string;
  temperature: string;
  condition: string;
} {
  console.log(`Getting current weather for ${city}`);
  return { city, temperature: "25°C", condition: "Sunny" };
}

// Simulated upload_post tool.
export async function uploadPost(
  title: string,
  description: string
): Promise<void> {
  const url = "https://node119.cs.colman.ac.il:4000/posts";
  const form = new FormData();

  // Prepare the JSON content.
  const postContent = JSON.stringify({ title, content: description });
  form.append("post", postContent, { contentType: "application/json" });

  // Adjust image path as needed.
  const imagePath = path.resolve(__dirname, "../image.jpeg");
  form.append("file", fs.createReadStream(imagePath), {
    filename: "image.jpeg",
    contentType: "image/jpeg",
  });

  try {
    const response = await axios.post(url, form, {
      headers: {
        ...form.getHeaders(),
        authorization: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
      },
      // Note: verify SSL settings as needed
      // eslint-disable-next-line @typescript-eslint/no-require-imports
      httpsAgent: new (require("https").Agent)({ rejectUnauthorized: false }),
    });
    console.log("Upload response:", response.data);
  } catch (error) {
    console.error("Error uploading post:", error);
  }
}
